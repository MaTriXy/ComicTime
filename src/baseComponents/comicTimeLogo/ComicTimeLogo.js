import React from "react";
import appLogo from "../images/logo.svg";
import { translateComponent } from "../../i18n/i18n";

const style = {
  backgroundColor: "#FFFFFF",
  borderRadius: "50%"
};

class ComicTimeLogo extends React.Component {
  render() {
    const { t } = this.props;
    return <img src={appLogo} style={style} alt={t("appLogo")} />;
  }
}

export const PureComicTimeLogo = ComicTimeLogo;
export default translateComponent(PureComicTimeLogo);
