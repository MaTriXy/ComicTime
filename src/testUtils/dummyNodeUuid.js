jest.mock("node-uuid", () => {
  return {
    v4() {
      return "uuid";
    }
  };
});
