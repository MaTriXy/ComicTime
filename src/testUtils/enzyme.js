import React from "react";
import Enzyme from "enzyme";
import { createSerializer } from "enzyme-to-json";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import { MuiThemeProvider } from "material-ui";
import comicTimeTheme from "../theme";

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));
Enzyme.configure({ adapter: new Adapter() });

export function mockHistory() {
  return {
    go: jest.fn(),
    goBack: jest.fn()
  };
}

export function translator(key) {
  return key;
}

export function mountWithMuyThemeProvider(component) {
  return Enzyme.mount(
    <MuiThemeProvider muiTheme={comicTimeTheme}>{component}</MuiThemeProvider>
  );
}

export function mountComponent(component) {
  return Enzyme.mount(component);
}

export function mountComponentAsJson(component) {
  return toJson(mountComponent(component));
}

export function shallowComponent(component) {
  return Enzyme.shallow(component);
}

export function shallowComponentAsJson(component) {
  return toJson(shallowComponent(component));
}

export function json(component) {
  return toJson(component);
}
