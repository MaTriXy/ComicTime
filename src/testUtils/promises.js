export async function flushPromises() {
  return new Promise(resolve => setImmediate(resolve));
}
