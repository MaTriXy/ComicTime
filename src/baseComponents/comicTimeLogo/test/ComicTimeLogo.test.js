jest.mock("../../../i18n/i18n");
import React from "react";
import { PureComicTimeLogo } from "../ComicTimeLogo";
import { mountComponentAsJson, translator } from "../../../testUtils/enzyme";

describe("ComicTimeLogo", () => {
  it("renders the application comicTimeLogo as an image", () => {
    const component = mountComponentAsJson(
      <PureComicTimeLogo t={translator} />
    );
    expect(component).toMatchSnapshot();
  });
});
