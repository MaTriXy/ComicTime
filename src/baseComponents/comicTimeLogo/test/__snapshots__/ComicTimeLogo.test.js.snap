// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComicTimeLogo renders the application comicTimeLogo as an image 1`] = `
<ComicTimeLogo
  t={[Function]}
>
  <img
    alt="appLogo"
    src="logo.svg"
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": "50%",
      }
    }
  />
</ComicTimeLogo>
`;
